### ISSUE TITLE
Support for other DB than ActiveAndroid

### ISSUE THREAD

**Issue by FrankDomburg** [smaugho/declex#81]
ActiveAndroid is nice, really nice. However, it appears not to support encryption directly.

Basing Declex on Realm (for instance), would allow transparent encryption both locally and on the server.

In this day and age, IMHO, a very basic requirement for many apps.

Could it be possible to either base directly on Realm, or to have an abstract interface with methods that need to be implemented to be able to connect to non-ActiveAndroid dbmanagers?

**Comment by smaugho** [comment]
Yes, in general the Model mechanism permits to attach it to any ORM. Right now DecleX uses it's own annotation to configure the model... of course, it have to use ActiveAndroid Annotations (for instance @Column over the fields) to be fully compatible with it. 

A first approach it is simply to permit to the programmer to specify what ORM he wants to use with the Local-DB interface (so Realm could be perfectly one of them). 

A more general one it is through plugins. DecleX inherits the Plugins mechanism from AndroidAnnotations, and it has been modified to permit more flexibility. So through one plug in, it could even override the default behavior of the DB-annotation for models, permitting with a simple "compile ... Realm-plugin" configure the framework to use Realm instead of ActiveAndroid.
